### PR TITLE
MR-2069: Remove old DraftSubmission/StateSubmission Types

### PR DESCRIPTION
**Pull request by rswerve**
## Summary

Draft for now.  Have one weird date behavior to fix and test, and some console.logs I'm still using.

### CODE REVIEW

File: services/app-web/src/pages/StateSubmission/ContractDetails/ContractDetails.tsx [hunk 1]
```diff
@@ -300,48 +301,57 @@ export const ContractDetails = ({
...
+        draftSubmission.contractDocuments = contractDocuments
 
         if (values.contractType === 'AMENDMENT') {
             const relatedToCovid = values.relatedToCovid19 === 'YES'
             const relatedToVaccine = relatedToCovid
                 ? values.relatedToVaccination === 'YES'
-                : null
+                : undefined
+
+            // const amendedOther = formatForApi(values.otherItemAmended)
```
Comment (macrael) [inline services/app-web/src/pages/StateSubmission/ContractDetails/ContractDetails.tsx:324]: some commented out code in here still

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx [hunk 2]
```diff
@@ -16,9 +16,10 @@ import styles from '../StateSubmissionForm.module.scss'
 
 import {
     Document,
-    DraftSubmission,
+    // DraftSubmission,
```
Comment (macrael) [inline services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx:19]: commented

File: services/app-web/src/testHelpers/apolloHelpers.tsx [hunk 3]
```diff
@@ -51,15 +54,15 @@ function mockValidCMSUser(): UserType {
     }
 }
 
-export function mockDraft(): DraftSubmission {
+export function mockDraft(): UnlockedHealthPlanFormDataType {
```
Comment (macrael) [inline services/app-web/src/testHelpers/apolloHelpers.tsx:57]: These mock form datas could be moved into the domain-mocks package so we can start to consolidate them into one place.

**Review by macrael — APPROVED** [review]
This is so exciting! It seems like outside contract details and rate details there weren't a ton of changes to be made to individual form pages. LMK if you want a rubber duck on the date issue

File: services/app-web/src/components/SubmissionSummarySection/SubmissionTypeSummarySection/SubmissionTypeSummarySection.tsx [hunk 4]
```diff
@@ -24,24 +26,26 @@ export const SubmissionTypeSummarySection = ({
     navigateTo,
     headerChildComponent,
     intiallySubmittedAt,
+    submissionName,
 }: SubmissionTypeSummarySectionProps): React.ReactElement => {
     const isPreviousSubmission = usePreviousSubmission()
     const programNames = statePrograms
         .filter((p) => submission.programIDs.includes(p.id))
         .map((p) => p.name)
+    const isSubmitted = submission.status === 'SUBMITTED'
```
Comment (haworku) [inline services/app-web/src/components/SubmissionSummarySection/SubmissionTypeSummarySection/SubmissionTypeSummarySection.tsx:35]: I see this `isSubmitted` being defined  in a couple files - can we just throw in a helper?

**Comment by rswerve** [comment]
> This is so exciting! It seems like outside contract details and rate details there weren't a ton of changes to be made to individual form pages. LMK if you want a rubber duck on the date issue

Thanks.  The date issue turned out to be some weird Firefox thing I need to write a bug for.  Chrome is fine.

File: services/app-web/src/components/SubmissionSummarySection/ContractDetailsSummarySection/ContractDetailsSummarySection.tsx [hunk 5]
```diff
@@ -13,17 +13,18 @@ import {
...
 import styles from '../SubmissionSummarySection.module.scss'
+import { HealthPlanFormDataType } from '../../../common-code/domain-models'
 
 export type ContractDetailsSummarySectionProps = {
-    submission: DraftSubmission | StateSubmission
+    submission: HealthPlanFormDataType
     navigateTo?: string
     documentDateLookupTable?: DocumentDateLookupTable
     isCMSUser?: boolean
+    submissionName?: string
```
[the remark this answers is not in the document]
Comment (rswerve) [reply services/app-web/src/components/SubmissionSummarySection/ContractDetailsSummarySection/ContractDetailsSummarySection.tsx:27]: Good catch.  Shouldn't be.

File: services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx [hunk 6]
```diff
@@ -393,7 +382,8 @@ export const SubmissionSummary = (): React.ReactElement => {
                 ) : null}
 
                 <SubmissionTypeSummarySection
-                    submission={submission}
+                    submission={packageData}
+                    submissionName={submissionName(packageData, statePrograms)}
```
Comment (haworku) [inline services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx:386]: maybe don't need to surface `submissionName` as a prop at all. looks like its a calculated value, we could calculate inside each XYZSummarySection

**Comment by haworku** [comment]
Everything LGTM so far. Appreciate this change for removing some hacky stuff we had going on

File: services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx (hunk 6 above)
Comment (macrael) [reply services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx:386]: It’s computed but requires the {program.id: program.nickname} mapping to compute, if we have to pass something down I’d rather it be the submissionName I think.

**Comment by macrael** [comment]
I suspect all your cypres errors are due to the mutation name being called changing. You need to fix what mutation you are waiting for on every click of Continue.

**Pull request merged**